doc/source/data/transforming-data.rst

```diff
@@ -1,286 +1,287 @@
-.. _transforming_data:
+.. _transforming-data:
 
 =================
-Transforming Data
+Transforming data
 =================
 
-Dataset transforms take in datasets and produce new datasets. For example, *map_batches*
-is a transform that applies a
-:ref:`user-defined function <transform_datasets_writing_udfs>` on each data record
-and returns a new dataset as the result. Dataset transforms can be composed to
-express a chain of computations.
-
---------
-Overview
---------
-
-There are two main types of supported transforms:
-
-* One-to-one: each input block will contribute to only one output
-  block, such as :meth:`ds.map_batches() <ray.data.Dataset.map_batches>`.
-* All-to-all: input blocks can contribute to multiple output blocks,
-  such as :meth:`ds.random_shuffle() <ray.data.Dataset.random_shuffle>`.
-
-.. list-table:: Common Ray Data transforms.
-   :header-rows: 1
-
-   * - Transform
-     - Type
-     - Description
-   * - :meth:`ds.map() <ray.data.Dataset.map>`
-     - One-to-one
-     - Apply a given function to individual data records.
-   * - :meth:`ds.map_batches() <ray.data.Dataset.map_batches>`
-     - One-to-one
-     - Apply a given function to batches of records.
-   * - :meth:`ds.repartition() <ray.data.Dataset.repartition>`
-     - All-to-all
-     - | Repartition the dataset into N blocks.
-   * - :meth:`ds.random_shuffle() <ray.data.Dataset.random_shuffle>`
-     - All-to-all
-     - | Randomly shuffle the dataset.
-   * -  :meth:`ds.groupby().\<agg\>() <ray.data.Dataset.groupby>`
-     - All-to-all
-     - | Group data by column and aggregate each group.
-   * -  :meth:`ds.groupby().map_groups() <ray.data.grouped_data.GroupedData.map_groups>`
-     - All-to-all
-     - | Group data by column and transform each group.
-
-.. _transform_datasets_writing_udfs:
-
---------------
-Map transforms
---------------
-
-Use ``map_batches`` to efficiently transform records in batches, or ``map`` to transform records individually:
+Transformations let you preprocess data and perform inference. This guide shows you
+how to:
 
-.. tab-set::
+* `Transform rows <#transforming-rows>`_
+* `Transform batches <#transforming-batches>`_
+* `Transform groups <#transforming-groups>`_
+* `Shuffle rows <#shuffling-rows>`_
```

Comment:
Group by is documented in "Transform groups". 

I don't think we should document aggregations because they don't seem relevant to our target use cases (batch inference and ML ingest). Seems like aggregations are a historical artifact of our original "Arrow on Ray" positioning?

Reply:
Shall we punt this to a separate page? In the short term, we could also leave it to pydoc only, similar to our preprocessors.

Reply:
ok that sgtm